cmd/minishift/cmd/config/config.go

```diff
@@ -100,6 +100,14 @@ var settings []Setting = []Setting{
 		callbacks:   []setFn{RequiresRestartMsg},
 	},
 	{
+		name: "hostfolders-automount",
```

Comment:
> This especially makes sense when minishift create is considered.

'minishift create' does address another problem primarily. It is mainly about clearly separating between an actual creation of the VM opposed to starting/stopping it. It also addresses the problem we have with options having to match on a restart, etc.